src/bindings.h

```diff
@@ -67,8 +67,15 @@ enum lxcfs_virt_t {
 
 	LXC_TYPE_SYS_DEVICES_SYSTEM_CPU_ONLINE,
 #define LXC_TYPE_SYS_DEVICES_SYSTEM_CPU_ONLINE_PATH "/sys/devices/system/cpu/online"
+	LXC_TYPE_END,
 };
 
+/* Macros below used to check the class from the file types above */
+#define LXCFS_TYPE_CGROUP(type) (type >= LXC_TYPE_CGDIR && type <= LXC_TYPE_CGFILE)
+#define LXCFS_TYPE_PROC(type) (type >= LXC_TYPE_PROC_MEMINFO && type <= LXC_TYPE_PROC_SLABINFO)
+#define LXCFS_TYPE_SYS(type) (type >= LXC_TYPE_SYS && type <= LXC_TYPE_SYS_DEVICES_SYSTEM_CPU_ONLINE)
+#define LXCFS_TYPE_OK(type) (type >= LXC_TYPE_CGDIR && type < LXC_TYPE_END)
```

Comment:
This is error prone as it depends on ordering in the enum. The ordering isn't enforced however. So there needs to be at least a comment on top of the enum explaining this. Also, we should explicitly number the enum. This also makes it harder to accidently break ordering requirements.

Reply:
Technically, right now it's also not correct to add new values in between the enum. Cause we can perform reloading of `liblxcfs` in between `open` and `close` of the file descriptor. Then because enum values shift, some condition checks may become incorrect. For example https://github.com/lxc/lxcfs/blob/master/src/sysfs_fuse.c#L372

Reply:
I also did give this some thought beforehand. I thought some ways to do this could be..
1. re-order the enum numbering explicitly granting space for future files. IE
```
enum lxcfs_virt_t { 
  LXC_TYPE_CGROUP_BASE = 0x00000,
...
  LXC_TYPE_CGROUP_MAX,

  LXC_TYPE_PROC_BASE = 0x10000,
...
  LXC_TYPE_PROC_MAX,

  LXC_TYPE_SYS_BASE ,
...
  LXC_TYPE_SYS_MAX = 0x20000,
}
```
2. extend `file_info` to include a 'class' parameter, ditch the defines, remove `file_info_type` in favour of `file_info_class`.

For 1. Obviously you'd basically have to go through every file type already configured and put them in the right class, but at that point ordering no longer matters as you're relying on a different set of metadata to work out what goes where..

Also you could hijack the unused `fileinfo->buf` pointer that is and change it to a `unsigned char abi_version[sizeof(char *)]` that you could check going forwards. This would let you conditionally and smartly extend the runtime ABI yet be able to deal with previous versions of the ABI changing.
This is _of course_ assuming *buf is zeroed on initialization currently (version 0 implictly).
This _also_ felt like a whole load of architectural work for what is essentially a bugfix!

For 2. Its much simpler to implement and prevents in the future having to make a headache again.  That however breaks the ABI in a manner which would not survive a reload so didn't imagine that would get any traction..

Reply:
Just also thought you could actually do `LXC_TYPE_BASE = 0x00000`, copy all the file types as-is, then have new namespaces for each type and defines for which you use going forwards. That gives you both the means to maintain `reload` compatibility by checking if the type is in a 'base' namespace or the given correct namespace.

Its a little janky but probably simpler than doing some abi extension.

Reply:
For now just leaving a comment is fine. We just want to fix the immediate bug. Reworking it to something better should be a separate step. But feel free to follow this up with a series.

Reply:
Added the requested comment above the ENUM indicating only appending to the bottom (above max).